File: AndroidProject/CollectorAgentMind/src/main/java/com/example/collectoragentmind/codelets/motor/LegActionCodelet.java
package com.example.collectoragentmind.codelets.motor;

import com.example.collectoragentmind.CollectorAgentMind;
import com.example.collectoragentmind.util.Vector2;

import br.unicamp.cst.core.entities.Codelet;
import br.unicamp.cst.core.entities.MemoryObject;

public class LegActionCodelet extends Codelet {

    /*
        Sends commands from the legsMO to Godot as signals
    */

    private CollectorAgentMind plugin;

    private MemoryObject legsMO;

    public LegActionCodelet() {
        plugin = CollectorAgentMind.getInstance();
    }

    @Override
    public void accessMemoryObjects() {
        legsMO = (MemoryObject) getInput("LEGS");
    }

    @Override
    public void calculateActivation() {

    }

    @Override
    public void proc() {
        Vector2 legAimPos = (Vector2) legsMO.getI();

        if (legAimPos == null) {
            plugin.forage();
        } else {
            plugin.goTo(legAimPos.x, legAimPos.y);
        }

        CollectorAgentMind.getInstance().updateValue(legsMO.getI().toString(), "LEGS");
    }
}

File: GodotKotlin/CoreModelTrail/src/main/kotlin/com/example/game/cst/codelets/motor/LegActionCodelet.kt
package com.example.game.cst.codelets.motor

import br.unicamp.cst.core.entities.Codelet
import br.unicamp.cst.core.entities.MemoryObject
import com.example.game.CollectorMindCommunicator
import godot.core.Vector2

class LegActionCodelet : Codelet() {
    /*
        Sends commands from the legsMO to Godot as signals
    */

    private var legsMO: MemoryObject? = null

    override fun accessMemoryObjects() {
        legsMO = getInput("LEGS") as MemoryObject
    }

    override fun calculateActivation() {}

    override fun proc() {
        val legAimPos: Vector2? = legsMO!!.i as Vector2?
        if (legAimPos == null) {
            CollectorMindCommunicator.instance.forage()
        } else {
            CollectorMindCommunicator.instance.goTo(legAimPos)
        }
    }
}